lib/jekyll-compose/arg_parser.rb

```diff
@@ -0,0 +1,27 @@
+class Jekyll::Compose::ArgParser
+  attr_reader :args, :options
+  def initialize(args, options)
+    @args = args
+    @options = options
+  end
+
+  def validate!
+    raise ArgumentError.new('You must specify a name.') if args.empty?
+  end
+
+  def type
+    type = options["type"] || Jekyll::Compose::DEFAULT_TYPE
+  end
+
+  def layout
+    layout = options["layout"] || Jekyll::Compose::DEFAULT_LAYOUT
+  end
+
+  def title
+    args[0]
```

Comment:
The previous implementation used args.shift, so this seemed equivalent. I'll use args.join that definitely seems better.